Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,283 @@
+        (i.e. scrollView is near the top of CampusMenuViewController)
+        The first operation is very simple, as visible by the "if" statement below. However, the second
+        (occurs in "else) is much more complicated because we have to split the automatic scroll operation
+        into two pieces: scrolling to the top of this view controller and scrolling to the target.  If we
+        don't do this, the tabBarStack will scroll unevenly with the rest of the scroll view.   */
+    @objc func manualScroll() {
+        let minOffset: CGFloat = 4
+        let currentPos = scrollView?.contentOffset.y ?? 0
+        let viewControllerPosOnParent = view.frame.origin.y
+        let viewPos = originalViews[tabBar.selectedSegmentIndex].frame.minY
+        let targetPos = viewPos - scrollOffset + viewControllerPosOnParent + minOffset
+        let topOfSelf = viewControllerPosOnParent - scrollOffset
+        let duration = animationDuration(from: currentPos, to: targetPos, withConst: 0.03)
+
+        isManualScrolling = true
+
+        if currentPos > topOfSelf {
+            UIView.animate(withDuration: duration, delay: 0, options: .curveEaseInOut) {
+                self.scrollView?.contentOffset.y = targetPos
+            }
+        } else {
+            // firstProp represents the proportion of animation time spent scrolling from currentPos to the top of self
+            // secondProp represents the proportion of animation time spent scrolling from the top of self to targetPos
+            let firstProp = fabs((currentPos) - topOfSelf) / fabs((currentPos) - targetPos)
+            let secondProp = fabs(topOfSelf - targetPos) / fabs((currentPos) - targetPos)
+            let firstDuration = duration * Double(firstProp)
+            let secondDuration = duration * Double(secondProp)
+
+            UIView.animate(withDuration: firstDuration, delay: 0, options: .curveEaseIn, animations: {
+                self.scrollView?.contentOffset.y = viewControllerPosOnParent - self.scrollOffset
+            }, completion: { completed in
+                if completed {
+                    UIView.animate(withDuration: secondDuration, delay: 0, options: .curveEaseOut) {
+                        self.scrollView?.contentOffset.y = targetPos
+                    }
+                }
+            })
+
+        }
+
+        isManualScrolling = false
+    }
+
+    /*  This function helps scroll animations look more proportional.  However, smaller distances should travel
+        slower, proportionally, compared to larger distances.  That's why this function is logarithmic rather
+        than linear.  If it were linear, shorter scrolls would look too short and choppy.   */
+    private func animationDuration(from cur: CGFloat, to target: CGFloat, withConst propConstant: CGFloat) -> Double {
+        let distance = fabs(cur - target)
+        return Double(log(distance) * log(distance) * propConstant) // Prop constant should be less than 0.1
```

Comment:
Can you explain why `propConstant` should be less than `0.1`?

Reply:
Because distances a scroll will travel are usually about 100-3000 pixels, ln(3000)^2 = 32.5 (which is actually log() in swift), this would be WAYY too long for an animation.  Setting prop constant less than 0.1, would make the animation a more reasonable number (3.25 at the most).

Reply:
Do you want me to explain this in my code or were you just asking on github?